### DISCUSSION TITLE
Is SQL Server Express edition supported (as instance to be monitored)?

### DISCUSSION THREAD

**Discussion by carehart** [trimble-oss/dba-dash#71] [General]
Quick Question: Would you say that you do or do not intend to support SQL Server Express editions? I'm having no luck connecting to a 2k8 Express edition. More below. I also don't find mention of it (either way) in the repo here.

First, nice looking too, thanks. Just learned of it today (via mention on Brent Ozar's blog post). Got it installed and connected to some servers.

But when I try to connect it to a 2008 Express edition instance, it never shows up in the tool...and your log shows a failure due to the instance not having the SQL Agent running: 
`Microsoft.SqlServer.Management.Smo.UnsupportedFeatureException: SQL Server Agent is not supported on this edition of SQL Server.`

And indeed one of the limitations of the Express edition is that the Agent is not supported (the service will not start). This is true even SQL Server 2019's implementation of Express

So do you feel dbadash does REQUIRE it, in order to connect at all? Might there be some config change I could make in the dbadash source connection that might allow it to proceed to at least connect and perhaps let is show SOME characteristics of the instance?

FWIW, I'll note that I AM indeed able to connect to the instance in question, via SSMS, from the machine on which I have implemented dbadash. And I I do run Activity Monitor from that also against that instance, using the same user I configured for dbadash to use for tha tinstance.

And I have some other SQL Server monitoring tools which work from this machine, monitoring that instance.  They all do at least connect to and show the instance, though some DO indicate how some feature of that monitoring tool may or does not work with an Express instance, sometimes specifically because of the lack of Agent support.

So bottom line, it seems that Agent support is mandatory for dbadash to connect. Or am I missing something?

Again, thanks for the tool. I will still be considering recommending it for others looking at SQL Server Monitoring solutions. And I do look forward to watching how it may evolve over time. I've also subscribed to [your youtube channel](https://www.youtube.com/channel/UCjACQ6uAfYPQJ2zydZxnAKg) to keep an eye on things.

Finally, feel free to let me know (and for the sake of other readers seeing this) whether you may prefer a question like this to be in the "issues" feature rather than in "discussions". Many github repos don't offer the "discussions" tab, so many folks never even notice it's an option to consider, when they may have something to discuss. :-)

**Comment by DavidWiseman** [comment]
Hi,  Thank you for your question!  I haven't actually tested it on Express edition.  It does work on 2005-2019, Azure DB, Azure Managed Instance. RDS,  Standard and Enterprise editions.  It doesn't rely on SQL Agent and I'm a little surprised that there are issues with Express edition.  I'm sure any problems will be easy to fix.  I'll try this out and get back to you.

I just opened the discussions section so people could have a place to ask questions that might not be a bug report for the app.  This question is fine under discussions though it might become an issue later.  There is also a slack channel #dbadash that can also be used to ask questions on the SQL Server Community slack.  https://dbatools.io/slack

**Comment by DavidWiseman** [comment]
I've tried 2008 Express and 2008R2 Express in my lab environment.  In both cases, I get an error with the Jobs collection.  I also get a problem with ServerExtraProperties collection on 2008R2 Express.  I'll get a fix for these in the next release.  

The instances show up in the app OK  - I just get some errors logged for these collections and everything else is working as normal.  
![image](https://user-images.githubusercontent.com/33157668/158265129-c2bcf3d0-d31c-4905-a49e-4e8843f9fd3f.png)

If you just added an instance to the collection you can click **Tags\Clear All** to get the instances to refresh in the app without restarting the GUI.  Let me know if the instances are still not showing up in the app.  

A workaround for the errors generated for the collection would be to disable the collection.  In "Exising Connections" on the config tool - scroll to the right and click "Schedule" for the instance.  Find the Jobs collection.  Uncheck "Default".  Replace the cron expression "0 0 * ? * *" with a blank string to disable the collection.  
Note: This wouldn't fix a problem with the instance not showing up in the app.  A fix should be available soon for the collection issue anyway.

**Comment by DavidWiseman** [comment]
@carehart There is a new release of DBA Dash that should fix the problems with SQL Express.  Let me know if this fixes your problem.
